geomstats/learning/frechet_mean.py

```diff
@@ -138,7 +139,13 @@ def _default_gradient_descent(
         sq_dists_between_iterates.append(sq_dist)
 
         var_is_0 = gs.isclose(var, 0.0)
-        sq_dist_is_small = gs.less_equal(sq_dist, epsilon * metric.dim)
+
+        metric_dim = metric.dim
+        if isinstance(metric, SRVMetric):
```

Comment:
I'm not very found of these special cases in the code... why do we need this?

Reply:
This is important since in the case of Discrete Curves the dimensionality of ambient encoding space is not infinite and hence has to be reset to a finite value, which we are trying to fix. @ninamiolane please correct me if am wrong.

Reply:
Yes, this is because the DiscreteCurves manifold does not have a set dimensionality (contrarily to the other manifolds), and we have set up its dimensionality as infinity (which would be the case if the curves were not discrete but continuous).

This is a bad design for DiscreteCurves class, bcs it has led to this bug in FrechetMean ---- which justifies this (agreed, hacky) fix.

Reply:
Do you see a way to improve the design in the future @ninamiolane? This situation looks very dangerous to me and will make `metric.dim` a non-reliable attribute.

For now, can't we hack this in a different way? For example, can't we use e.g. the information about `point_type` to get `metric_dim` from the `tangent_mean` shape? This way it would be applicable to all (future) cases where `dim` is infinite.

Reply:
@shubhamtalbar96, do you see a way of doing this in a different way or we keep this like this for now?

If not, then let's just update `predict_labels` following #1530 and merge. (Or is there anything else missing?)